### ISSUE TITLE
cannot use intermediate position and tilt/ventilaton

### ISSUE THREAD

**Issue by andy-st** [W00D00/home-assistant-elero#10]
Accoding to the Elero Stick documentation, there are 5 commands:

![image](https://user-images.githubusercontent.com/57436486/68615335-2559d700-04c3-11ea-822b-1a95a18ff5c0.png)

I've defined a shutter like this:

```
    - platform: elero
      covers:
          rolloarbeitszimmer:
              serial_number: A503EN6C
              name: Arbeitszimmer
              channel: 1
              device_class: roller shutter
              supported_features:
                  - up
                  - down
                  - stop
                  - set_position
                  - set_tilt_position
```

It looks like this:

![image](https://user-images.githubusercontent.com/57436486/68615493-7b2e7f00-04c3-11ea-9926-ff67c92ef2fd.png)

But if I use the sliders for position or tilt, just nothing happens.

I can then find in the log file:
2019-11-11 19:39:55 WARNING (SyncWorker_14) [custom_components.elero.cover] Elero - transmitter: 'A503EN6C' ch: '1' The set cover tilt position function is not implemented yet.

So directly I can only click on “open”, “stop” and “close”.
I would expect that there is also something to click on “move to intermediate position” and “move to ventilation position”. Both commands should work without giving an argument to this method. I mean, the Elero shutters only can execute e.g. “move to ventilation position”, but they do not support something like “move 30%”. The both positions “intermediate” and “ventilation” can be programmed separately and are stored in the shutter itself.

It would be great if there would a way how you could send the shutters to the predefined "intermediate position" and "tilt/ventilation". :-)

**Comment by W00D00** [comment]
Hi @andy-st,

The intermediate and ventilation functions are working well for me. 
Based on your configuration, you have to add the 'open_tilt' and the 'close_tilt' functions to your config. The 'set_position' and the 'set_tilt_position' features alone are not enough. However, in the latest version of the Elero lib on the Github these two slider functions are not implemented. Please give me some time and I will implement a solution to the sliders.

Cover 'Position' and 'Tilt position' Sliders

Unfortunately, by default, the Position slider is not configurable on a cover so, the 'step' of the slider either. Thus, the 'set_position' and the 'set_tilt_position' functions are not usable. Another problem that the Elero devices are not supporting these functions.

For the Elero 'intermediate' function use the 'open_tilt' HA function and the Elero 'ventilation' function use the 'close_tilt' HA function.

Nevertheless, these controls are shown and useable only if the pop-up window of the given cover is open.

Alternative methods for the Elero 'intermediate' and the 'ventilation' functions:

1. [Call a Service](https://www.home-assistant.io/docs/scripts/service-calls/)


entities:
  - name: Intermediate
    service: cover.close_cover_tilt
    service_data:
      entity_id: cover.all_cover_group
    type: call-service
  - name: Ventilation
    service: cover.open_cover_tilt
    service_data:
      entity_id: cover.all_cover_group
    type: call-service


2. An ['input_number'](https://www.home-assistant.io/integrations/input_number/) slider with automation.

input_number:
    diningroom_set_position:
        name: Position
        mode: slider
        initial: 0
        min: 0
        max: 100
        step: 25

automation:
  - alias: diningroom_set_position
    trigger:
        platform: numeric_state
        entity_id: input_number.diningroom_set_position
        to: 25
    action:
        - service: cover.close_cover_tilt
          entity_id:
            - cover.diningroom


3. An ['input_select'](https://www.home-assistant.io/integrations/input_select/) Scene with automation.

input_select:
    scene_diningroom:
        name: Scene
        options:
            - open
            - close
            - stop
            - intermediate
            - ventilation

automation:
  - alias: Diningroom scene
    trigger:
      platform: state
      entity_id: input_select.scene_diningroom
      to: intermediate
    action:
        - service: cover.close_cover_tilt
          entity_id:
            - cover.diningroom

I hope these will help you until I will deploy a new version of the Elero lib with new slider support.

Thanks

**Comment by andy-st** [comment]
Hello @W00D00,

wow, thank you so much for your detailled description! 👍 

I've now added all functions like this:

```
    - platform: elero
      covers:
          rolloarbeitszimmer:
              serial_number: A503EN6C
              name: Arbeitszimmer
              channel: 1
              device_class: roller shutter
              supported_features:
                  - up
                  - down
                  - stop
                  - set_position
                  - open_tilt
                  - close_tilt
                  - stop_tilt
                  - set_tilt_position
```

> For the Elero 'intermediate' function use the 'open_tilt' HA function and the Elero 'ventilation' function use the 'close_tilt' HA function. 
Nevertheless, these controls are shown and useable only if the pop-up window of the given cover is open.

So I understood that I should use
- intermediate --> open_tilt
- ventilation --> close_tilt

For me, it is still not working. :-( Here comes an example:

The shutter is opened and I open the pop-up window:

![image](https://user-images.githubusercontent.com/57436486/69239776-48f3e000-0b9b-11ea-9d7c-1d0121524de4.png)

As you can see I can click only on "Close Tilt" ( = ventilation), but not on "Open Tilt" ( = intermediate)?
Anyways, when I click on "Close Tilt" the shutter is moving to the intemediate (!) position, but I still cannot click "Open Tilt":

![image](https://user-images.githubusercontent.com/57436486/69239986-c881af00-0b9b-11ea-96eb-5c3cdcafcfc0.png)

So to summarize:
- "Close Tilt" moves to intermediate position and not to the ventilation position. 
- "Open Tilt" I can never click, it is always grey-out in the pop-up window.

I am using also a simple Elero UniTec-868 remote control and this is working fine:
- 2x clicks on "Open / Up" then the shutter moves to ventialation
- 2x clicks on "Close / Down" then the shutter moves to the intermediate position

Do you have idea what could be wrong?
Is my understanding correct, that it should work with the two arrows in the pop-up window before I start working with the services / input number / input select / ...?

Thanks a ton again!

**Comment by andy-st** [comment]
I did one more test with the development tools of HA:

I tried to call the following:

![image](https://user-images.githubusercontent.com/57436486/69240632-35497900-0b9d-11ea-80d4-f9809b95bbab.png)

-> then the shutter is moved to "ventilation". 

When calling "cover.close_cover_tilt" instead there, then the shutter is moved to "intermediate". 

So for me ventilation / intermediate are mixed up (in reference to your description), but this would not be any problem for me! :-)

But anyways, I've no idea why I cannot click "open_cover_tilt" in the pop-up window of a shutter. :-(

Perhaps I'll go on wit the services / input numbers / input selects anyways in the next days (as far as I understand there I could replace intermediate / ventilation quite easily as it is working here the other way round....).

**Comment by W00D00** [comment]
Hi @andy-st,

Great :) and very it is interesting.

1. Intermediate and Ventilation are not working.
    No idea, why it does not work for you.

2. The Intermediate and Ventilation are mixed up.
    It is not a big deal to change these functions. First, we need to verify that they are really in wrong order.

What do you think, we can discuss your issues online somehow? That would be more efficient than writing here. If yes, please share your skype name or facebook messenger link or something online messaging tool with me via mail istvan.szirtes@gmail.com and I will contact you. It would be more useful to see your HA install and config online via VNC maybe.

Thanks,

**Comment by andy-st** [comment]
Hello @W00D00,

thank you for your feedback. I am very sorry, but unfortunately I am nearly never online when I am at home (because of my job and my family), so I would be very difficult to "meet" you live online. Also my private computer is not equipped with audio / headset / ...
I know Skype etc., but I am only using these tools in my job and business computers, but with them I cannot access my private HA. :-(

I also already thought about getting Nabu Casa Remote UI working, then it would be easier also to have such a meeting even when I am not at home, but then it would not be able to give feedback what happens in my house...

So I think it would be too difficult to organize an online meeting and find a time slot where we both would be available.
But I would be very happy if I could support you in doing tests, creating log files, testing test / debug versions or similar. In this case, please just let me know what I could do for you. 

I am very sorry for that.

**Comment by andy-st** [comment]
Thank you again @W00D00 for the effort and time you are spending for this library! You are my Home Assistant hero! :-)

I did a short test and feel, some things are still not absolutely correct.

Here the steps:

1. The shutter is open:
![image](https://user-images.githubusercontent.com/57436486/70544552-8c26ea80-1b6c-11ea-9d9d-067eccce5966.png)

2. I move the position slider zu 75%. According to the documentation the shutter should now move to the VENTILATION position:

![image](https://user-images.githubusercontent.com/57436486/70544651-b678a800-1b6c-11ea-98b1-10951d5a1790.png)

But this is not the case - my shutter is moving to the INTERMEDIATE positon.

3. A few seconds later the GUI is updated automatically and the slider jumps to the 25% position:

![image](https://user-images.githubusercontent.com/57436486/70544762-e9bb3700-1b6c-11ea-876c-35448bc0f8ac.png)

So as far as I understand, the shutter now reports correctly, to be in INTERMEDIATE status. 

The other way round it is quite similar: 

1. The shutter is opened (100%).
2. I move the slider to 25% (INTERMEDIATE), but then the shutter is moved to VENTILATION and the slider moves automatically after a few seconds to 75%.

So the command and the status does not match to each other.

I think it is not important for the commands, but my shutters are defined in the following way, that INTERMEDIATE position is around 40 .. 60% and VENTILATION is around 5%. The setting, which position the INTERMEDIATE or the VENTILATION command does mean, can be set up in the shutter itself (with the normal hardware remote control). So everybody is free to define what INTERMEDIATE and VENTILATION does mean. So there might be people like me where VENTILATION position < INTERMEDIATE position, but you could also set it up that VENTILATION position > INTERMEDIATE position.

**Comment by W00D00** [comment]
Hi @andy-st,

Please try the new version of the ELero lib. I did some modification based on your request. 

What you have mentioned as a free setup for the vent and int positions, you can do it with my 2. alternative "An 'input_number' slider with automation." solution instead of the official cover position slider. Please see the readme.

I am waiting for your feedback.
Thanks,

**Comment by andy-st** [comment]
@W00D00, thank you so much, tilt / intermediate work brilliant now!
